### PR TITLE
add gvfs generic and versions

### LINKED ISSUE #1649: gvfs
While gvfs/1.24.1 is identified and has a version in issue-712-EA, gvfs/1.22.2 doesnt have the right version.

[Here](https://download.gnome.org/sources/gvfs/) is a list of available versions.

### PR DESCRIPTION
**Pull request by mmorel-35**
fix #1649

### CODE REVIEW

**Comment by mmorel-35** [comment]
I've seen more explicit logs !

**Comment by jaydiablo** [comment]
Not sure why the php-cs-fixer step failed in the build, since you're not modifying any php files.

You could try running this command:

`php vendor/bin/php-cs-fixer fix`

And see if it fixes anything, then add those chnages to the commit.

**Comment by mmorel-35** [comment]
Here is what I get : 

<pre>
Loaded config default from "/cygdrive/d/external-source/browscap/.php_cs".
PHP Fatal error:  Uncaught Error: Call to undefined function Symfony\Component\Filesystem\ctype_alpha() in /cygdrive/d/external-source/browscap/vendor/symfony/filesystem/Filesystem.php:600
Stack trace:
#0 /cygdrive/d/external-source/browscap/vendor/friendsofphp/php-cs-fixer/src/Console/ConfigurationResolver.php(286): Symfony\Component\Filesystem\Filesystem->isAbsolutePath('.php_cs.cache')
#1 /cygdrive/d/external-source/browscap/vendor/friendsofphp/php-cs-fixer/src/Console/ConfigurationResolver.php(197): PhpCsFixer\Console\ConfigurationResolver->getDirectory()
#2 /cygdrive/d/external-source/browscap/vendor/friendsofphp/php-cs-fixer/src/Console/Command/FixCommand.php(199): PhpCsFixer\Console\ConfigurationResolver->getCacheManager()
#3 /cygdrive/d/external-source/browscap/vendor/symfony/console/Command/Command.php(262): PhpCsFixer\Console\Command\FixCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 /cygdrive/d/external-source/browscap/vendor/symfony/cons in /cygdrive/d/external-source/browscap/vendor/symfony/filesystem/Filesystem.php on line 600
</pre>

**Comment by mmorel-35** [comment]
Here is what I get : 

<pre>
Loaded config default from "browscap/.php_cs".
PHP Fatal error:  Uncaught Error: Call to undefined function Symfony\Component\Filesystem\ctype_alpha() in browscap/vendor/symfony/filesystem/Filesystem.php:600
Stack trace:
#0 browscap/vendor/friendsofphp/php-cs-fixer/src/Console/ConfigurationResolver.php(286): Symfony\Component\Filesystem\Filesystem->isAbsolutePath('.php_cs.cache')
#1 browscap/vendor/friendsofphp/php-cs-fixer/src/Console/ConfigurationResolver.php(197): PhpCsFixer\Console\ConfigurationResolver->getDirectory()
#2 browscap/vendor/friendsofphp/php-cs-fixer/src/Console/Command/FixCommand.php(199): PhpCsFixer\Console\ConfigurationResolver->getCacheManager()
#3 browscap/vendor/symfony/console/Command/Command.php(262): PhpCsFixer\Console\Command\FixCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 browscap/vendor/symfony/cons in browscap/vendor/symfony/filesystem/Filesystem.php on line 600
</pre>

**Comment by mmorel-35** [comment]
I just made a pull request from your master to mine and then created my branch. Here is what the build is saying : 
<pre>
Files that were not fixed due to errors reported during fixing:
   1) /home/travis/build/browscap/browscap/src/Browscap/Coverage/Processor.php
                                       
        [RuntimeException]             
        Index invalid or out of range  
                                       
The command "php -n -d memory_limit=768M vendor/bin/php-cs-fixer fix --dry-run -vv" exited with 72.
</pre>

**Comment by mimmi20** [comment]
Please force the cs-fixer to use version 2.7.1. The version 2.7.2 breaks our build.

**Comment by jaydiablo** [comment]
For me, 2.7.1 just skips everything, not sure why, unless that's the cache:

>php -n -d memory_limit=768M vendor/bin/php-cs-fixer fix --dry-run -vvv
>Loaded config default from "/Users/jasonklehr/Sites/browscap/.php_cs".
>Using cache file ".php_cs.cache".
>SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
>Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error
>
>Checked all files in 0.054 seconds, 8.000 MB memory used

Ah, yeah, if I remove the cache file it does run properly:

>php -n -d memory_limit=768M vendor/bin/php-cs-fixer fix --dry-run -vvv
>Loaded config default from "/Users/jasonklehr/Sites/browscap/.php_cs".
>.....................................................................................................
>Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error
>
>Checked all files in 5.478 seconds, 14.000 MB memory used

@mmorel-35 generally you shouldn't run `composer update` unless you're trying to update dependencies.  `composer install` is better to run as it pulls down the dependenices as they are defined in the `composer.lock` file. In this case, `php-cs-fixer` 2.7.1 will be installed if you run `composer install` but if you run `composer update` it'll pull down the latest 2.7.x version, which is 2.7.2.

**Comment by jaydiablo** [comment]
That said, it seems that our travis build runs `composer update`, so perhaps we should change that so that it uses the `composer.lock` file.

I'll make a separate issue about that.

**Comment by mmorel-35** [comment]
What needs to be on my side after you fixed the composer problem to allow the merge of my branch?

**Comment by jaydiablo** [comment]
If/when #1662 gets merged, you'll need to rebase this branch, I usually do that like so:

```
git checkout master
git fetch browscap
git rebase browscap/master
git checkout issue-1649
git rebase master
```

Then you'll just need to push/sync with github and that should trigger a new build.

**Comment by mimmi20** [comment]
I opened an issue there : FriendsOfPHP/PHP-CS-Fixer#3202

**Comment by jaydiablo** [comment]
@mmorel-35 the issue with php-cs-fixer should be resolved if you rebase this branch and push it (you may have to force the push).

**Comment by mmorel-35** [comment]
sorry to ask but I have this : 
<pre>
git fetch 'http://github.com/browscap/browscap.git'
warning: redirecting to https://github.com/browscap/browscap.git/
From http://github.com/browscap/browscap
 * branch                HEAD       -> FETCH_HEAD
</pre>

then 
<pre>
git rebase
Display all 53 possibilities? (y or n)
5021                6000                6018
5022                6001                6019
5023                6002                6020
5024                6003                6021
5025                6004                6022
5026                6005                6023
5027                6006                6024
5028                6007                6025
5029                6008                6026
5030                6009                FETCH_HEAD
5031                6010                HEAD
5032                6011                issue-1653
5033                6012                master
5034                6013                origin/HEAD
5035                6014                origin/issue-1653
5036                6015                origin/issue-1656
5037                6016                origin/master
5038                6017
</pre>

and in any case
<pre>
git rebase master
Current branch master is up to date.
</pre>
or
<pre>
git rebase origin/master
Current branch master is up to date.
</pre>

**Comment by mimmi20** [comment]
@mmorel-35 Please revert your rebase or at least your last commit. There is something wrong with the last commit.

Your rebase command must be `git rebase browscap/master`. After the rebase you habe to commit it with force (`git push -f`).

File: tests/issues/issue-1649.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            'RenderingEngine_Name' => 'unknown',
+            'RenderingEngine_Version' => 'unknown',
+            'RenderingEngine_Maker' => 'unknown',
+        ],
+        'lite' => false,
+        'standard' => true,
+        'full' => true,
+    ],
+    'issue-1649-B' => [
+        'ua' => 'gvfs/1.36.2',
```
Comment (jaydiablo) [inline tests/issues/issue-1649.php]: Shouldn't this parse as 1.36?

If you're adding these to try to add coverage for the `generic` file, I generally try not to worry about those files, unless the agent has no version info in it. The issue here is that when 1.36 is released, and somebody adds the version to the file, this test will fail.

Comment (mmorel-35) [reply tests/issues/issue-1649.php]: how do you propose to cover generic ?

Comment (jaydiablo) [reply tests/issues/issue-1649.php]: You'll find that most of the generic files aren't covered for this reason.

We **could** add fake agents to the tests that the generic files would pick up, but that's a lot of maintenance, overhead, time for little value (IMO).

As I mentioned in #1670, we do have a file where we've been storing "fake" agents that do get picked up by generic files, that will increase coverage of the generic files, but that's typically when a UA is found that shouldn't be parsed as it looks (Say, Chrome on Windows 98 for example. We know that Chrome never supported Windows 98, so I've added such agents to that file to **prevent** that platform from being added to the Chrome files). If there was a version of an app/browser that was never released (and never will be), then I think that sort of agent could be added to that invalid-versions file, but I wouldn't add "might be released in the future" versions, as it's just more work for when that version is actually released and added to the files.

**Pull request merged**